### PR TITLE
Push notifications

### PR DESCRIPTION
**Pull request by nplasterer**
Android part of https://github.com/xmtp/xmtp-react-native/issues/22

https://user-images.githubusercontent.com/3522670/236349849-042ff434-49dd-45b8-a57b-b538575a072a.mp4

Will do iOS in a follow up.

### CODE REVIEW

File: example/src/PushController.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      },
+      // (required) Called when a remote or local notification is opened or received
+      onNotification(notification: { finish: (arg0: any) => void; }) {
+        console.log("NOTIFICATION:", notification);
+        // process the notification here
+        // required on iOS only
+        notification.finish(PushNotificationIOS.FetchResult.NoData);
+      },
+      // Android only
+      senderID: "609788839593",
```
Comment (nakajima) [inline example/src/PushController.tsx]: What's this number mean?

File: example/android/app/google-services.json
Comment (nakajima) [inline example/android/app/google-services.json:1]: Might want to fill this in with more generic stuff or just have this be a blank file with instructions on how to get your own google-services.json

File: example/src/PushController.tsx (hunk 1 above)
Comment (nplasterer) [reply example/src/PushController.tsx]: It's a firebase ID. It was required in the docs but It might not be needed anymore.

Comment (nakajima) [reply example/src/PushController.tsx]: Gotcha, might want to make that something like "YOUR FIREBASE ID HERE"

File: ios/XMTPModule.swift [hunk 2]
```diff
@@ -209,6 +209,18 @@ public class XMTPModule: Module {
...
+            // TODO
+        }
+
+        Function("subscribePushTopics") { (topics: [String]) in
+            // TODO
+        }
+
+        AsyncFunction("decodeMessage") { (topic: String, encryptedMessage: String, conversationID: String?) in
+            // TODO
+        }
```
Comment (nplasterer) [inline ios/XMTPModule.swift:223]: The spacing looks normal on in my VSCode and XCode so not sure why it looks like this here.

File: android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt [hunk 3]
```diff
@@ -92,7 +95,7 @@ class XMTPModule : Module() {
         Name("XMTP")
         Events("sign", "authed", "conversation", "message")
 
-        Function("address") {
+        Function("address") { clientAddress: String ->
```
Comment (nplasterer) [inline android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt:98]: App wouldn't compile unless I added these but haven't hooked up the testing stuff to actually use them.

**Review by jhaaaa — APPROVED** [review]
Thank you for this amazing engineering, @nplasterer - and also for taking the time to update the README! So grateful -- LGTM!

File: example/src/ConversationListView.tsx [hunk 4]
```diff
@@ -18,6 +18,7 @@ export default function ConversationListView({
 
   async function refreshConversations() {
     const conversations = await client.conversations.list();
+    XMTPPush.subscribe(conversations.map((c) => c.topic));
```
Comment (nplasterer) [inline example/src/ConversationListView.tsx:21]: It only subscribes once you refresh the list view. Not super great at react couldn't figure out how to get it to subscribe on streaming conversations but this is a minor detail and we can improve it later.

File: ios/XMTPModule.swift (hunk 2 above)
Comment (nakajima) [reply ios/XMTPModule.swift:223]: tabs v spaces

**Review by nakajima — APPROVED** [review]

**Pull request merged**